translate.js

```diff
@@ -18,9 +23,17 @@ function versionToSemver (version) {
   if (version.indexOf('0.1.3-0') !== -1) {
     return '0.1.3';
   }
+  if (version.indexOf('0.3.4-0') !== -1) {
+    return '0.3.4-nightly';
+  }
   if (version.indexOf('0.3.5-0') !== -1) {
     return '0.3.5';
   }
+  // This parses the obsolete nightly style where the date can have leading zeroes.
+  const nightlyParsed = version.match(/^([0-9]+\.[0-9]+\.[0-9]+)-nightly\.([0-9]+)\.0?([1-9])\.0?([1-9])(.*)$/);
```

Comment:
The `([0-9]+)\.0?([1-9])\.0?([1-9])(.*)` does not look quite right to me. Or am I missing something?

It looks to me like it would not accept two-digit month and day values that don't start with `0`. Like `12` or `27`. The test does seem to parse `2016.08.27` correctly though. Why?
- Is it because it pulls the final `7` into the `.*` part?
- Will it accept the `12` in `0.4.8-nightly.2016.12.16+commit.af8bc1c9.Emscripten.clang`?

Reply:
I think you can fix it by doing this instead:
```javascript
const nightlyParsed = version.match(/^([0-9]+\.[0-9]+\.[0-9]+)-nightly\.([0-9]+)\.0?([1-9]|1[012])\.0?(3[01]+|[12]?[0-9])(.*)$/);
```
So the one and two-digits days and months (excluding leading zeroes) are included.

Reply:
@r0qs Do you want to take over reviewing this?

Reply:
@cameel I took a look in this PR, and my suggestion actually does not work for cases like: `2018.2.32`, which would wrongly match: 2018, 2, and 3 instead of 32. Although 32 is not a valid day, the idea here is not to validate the date but only to match broken versions that contain the leading zeros. If a match does not happen, it will just return the version, assuming it is semver compatible.

So I changed @axic regex slightly to support cases like: `2016.12.06`